File: Sort/src/sort/BubbleSort.java
package sort;

/**
 * Bubble sort (冒泡排序)
 * 
 * @author fantasy
 */
public class BubbleSort {

    public static void bubbleSort(int[] data) {

        for (int i = 0; i < data.length; i++) {
            for (int j = i + 1; j < data.length; j++) {
                if (data[i] < data[j]) {
                    int temp = data[i];
                    data[i] = data[j];
                    data[j] = temp;
                }
            }
        }
    }
}

File: SortJS/BubbleSort.js
/**
 *
 * BubbleSort
 *
 * @description
 * @author Fantasy <fantasyshao@icloud.com>
 * @create 2014-09-21
 * @update 2014-10-02
 */


module.exports = function (arr) {

    var numElements = arr.dataStore.length;

    for (var i = numElements; i >= 2; --i) {
        for (var j = 0; j < i - 1; ++j) {
            if (arr.dataStore[j] > arr.dataStore[j + 1]) {
                arr.swap(arr.dataStore, j , j+1);
            }
        }
    }

    return arr.dataStore;
};